edx_api/enrollments/__init__.py

```diff
@@ -36,6 +46,56 @@ def get_student_enrollments(self):
         resp.raise_for_status()
         return Enrollments(resp.json())
 
+    def get_enrollments(self, course_id=None, username=None, usernames=None):
+        """
+        List all course enrollments.
+
+        Args:
+            course_id (str, optional): If used enrollments will be filtered to the specified course id.
+            username (str, optional): Enrollments will be filtered to this specific username.
+            usernames (list, optional): List of usernames to filter enrollments.
+        """
+        params = {'page_size': 1}
+        if course_id is not None:
+            params['course_id'] = course_id
+        if username is not None:
+            params['username'] = username
+        if usernames is not None:
+            params['username'] = ','.join(params.get('username', []) + usernames)
+
+        resp = self._get_list_request(params)
+        return Enrollments(resp.json()['results'])
+
+    def get_enrollments_by_course_id(self, course_id):
```

Comment:
These seem a bit redundant when the filters can be specified in the `get_enrollments()` function no?

Reply:
They are. But since this is a client API I thought it would be handy to also create methods to narrow the parameters requirements.
That way the code can be more readable when filtering courses only by course id or username.
But these are not really needed for functionality.